src/java.base/share/classes/java/net/Inet4Address.java

```diff
@@ -134,6 +137,64 @@ class Inet4Address extends InetAddress {
         holder().originalHostName = hostName;
     }
 
+    /**
+     * Creates an {@code Inet4Address} based on the provided textual representation of
+     * an IPv4 address.
+     * <p>The following IPv4 address {@linkplain Inet4Address##format
+     * textual representations} are supported by this method:
+     * {@snippet :
+     *  // Dotted-decimal 'd.d.d.d' form with four part address literal
+     *  Inet4Address.ofLiteral("7.08.9.010") ==> /7.8.9.10
+     *
+     *  // Dotted-decimal 'd.d.d' form with three part address literal,
+     *  // the last part is placed in the right most two bytes
+     *  // of the constructed address
+     *  Inet4Address.ofLiteral("127.0.257") ==> /127.0.1.1
+     *
+     *  // Dotted-decimal 'd.d' form with two part address literal,
+     *  // the last part is placed in the right most three bytes
+     *  // of the constructed address
+     *  Inet4Address.ofLiteral("127.257") ==> /127.0.1.1
+     *
+     *  // 'd' form with one decimal value that is stored directly in
+     *  // the constructed address bytes without any rearrangement
+     *  Inet4Address.ofLiteral("02130706689") ==> /127.0.1.1
+     * }
+     * <p>If the provided address literal cannot represent a valid IPv4 address an
+     * {@code IllegalArgumentException} is thrown.
+     * <p>This method doesn't block, i.e. no reverse lookup is performed.
+     *
+     * @param ipv4AddressLiteral the textual representation of an IPv4 address.
+     * @return an {@link Inet4Address} object with no hostname set, and constructed
+     *         from the IPv4 address literal.
+     * @throws IllegalArgumentException if the {@code ipv4AddressLiteral} cannot be
+     *         parsed as an IPv4 address literal.
+     * @throws NullPointerException if the {@code ipv4AddressLiteral} is {@code null}.
+     */
+    public static Inet4Address ofLiteral(String ipv4AddressLiteral) {
+        Objects.requireNonNull(ipv4AddressLiteral);
+        return parseAddressString(ipv4AddressLiteral, true);
+    }
+
+    /**
+     * Parses string with an IPv4 address literal.
+     * If string doesn't contain a valid literal - null is returned.
```

Comment:
This sentence is a bit confusing, given the implementation details of this method. I think `null` is only returned if `throwIAE` is false?